godot-core/src/builtin/array.rs

```diff
@@ -185,61 +282,189 @@ impl Array {
         let end = end.min(len);
         let step = step.unwrap_or(1);
         assert!(step != 0);
-        self.as_inner()
-            .slice(to_i64(begin), to_i64(end), step.try_into().unwrap(), true)
+        let slice: Array =
+            self.as_inner()
+                .slice(to_i64(begin), to_i64(end), step.try_into().unwrap(), true);
+        // SAFETY: slice() returns a typed array with the same type as Self
+        unsafe { slice.assume_type() }
+    }
+
+    /// Appends another array at the end of this array. Equivalent of `append_array` in GDScript.
+    pub fn extend_array(&mut self, other: TypedArray<T>) {
+        // SAFETY: Read-only arrays are covariant: conversion to a variant array is fine as long as
+        // we don't insert values into it afterwards, and `append_array()` doesn't do that.
+        let other: Array = unsafe { other.assume_type() };
+        self.as_inner().append_array(other);
+    }
+
+    /// Returns the runtime type info of this array.
+    fn type_info(&self) -> TypeInfo {
+        let variant_type =
+            VariantType::from_sys(self.as_inner().get_typed_builtin().try_into().unwrap());
```

Comment:
Could you replace `try_into().unwrap()` with `as i32`? We're not really benefitting from the check (the difference between `i64` vs. `i32` will never be relevant for our couple of ords), and it's more expressive.

Nice that you use a dedicated `TypeInfo` struct, by the way 🙂

Reply:
Done.